--- a/internal/pkg/build/sources/conveyorPacker_debootstrap.go
+++ b/internal/pkg/build/sources/conveyorPacker_debootstrap.go
@@ -185,6 +185,17 @@ func (cp *DebootstrapConveyorPacker) Get
 	if os.Getenv("DEB_GPG_KEYRING") != ""{
 		cmd_line = append(cmd_line,`--keyring=`+os.Getenv("DEB_GPG_KEYRING"))
 	}
+
+	if os.Getenv("SING_DEBOOTSTRAP_OPTS") != ""{
+		debootstrap_opts := strings.Fields(os.Getenv("SING_DEBOOTSTRAP_OPTS"))
+		for _, deb_opt := range debootstrap_opts {
+			cmd_line = append(cmd_line, deb_opt)
+			if strings.Contains(deb_opt, "--variant"){
+				cmd_line = cmd_line[1:]
+			}
+		}
+	}
+
 	cmd_line = append(cmd_line,`--arch=`+runtime.GOARCH)
 	cmd_line = append(cmd_line,cp.osversion)
 	cmd_line = append(cmd_line,cp.b.RootfsPath)
